projects/rocprofiler-sdk/source/include/rocprofiler-sdk/context.h

```diff
@@ -68,6 +69,40 @@ rocprofiler_start_context(rocprofiler_context_id_t context_id) ROCPROFILER_API;
 rocprofiler_status_t
 rocprofiler_stop_context(rocprofiler_context_id_t context_id) ROCPROFILER_API;
 
+/**
+ * @brief Priority service callback function.
+ *
+ * @param [in] other_client_id Other clients rocprofiler_client_id_t
+ * @param [in] current_context_id Current clients context ID
+ * @param [in] callback_data User data provided when configuring the callback tracing service
+ * @return ::rocprofiler_status_t
+ */
+typedef rocprofiler_status_t (*rocprofiler_callback_context_priority_service_t)(
+    rocprofiler_client_id_t  other_client_id,
+    rocprofiler_context_id_t current_context_id,
+    void*                    callback_data) ROCPROFILER_NONNULL(3);
```

Comment:
We need to provide more information here. Including, but not limited to, which services are conflicting. Although this may be surfaced through a "context service query" API

```suggestion
typedef rocprofiler_status_t (*rocprofiler_callback_context_priority_service_t)(
    const rocprofiler_client_id_t*  other_client_id,
    const rocprofiler_context_id_t current_context_id*,
    void*                    callback_data) ROCPROFILER_NONNULL(3);
```